src/Illuminate/Events/EventHooks.php

```
@@ -0,0 +1,627 @@
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function aggregateHierarchicalCallbacks(string $hook, string $event, array $payload): array
+    {
+        $this->validateHook($hook);
+
+        return $this->cache['hierarchical_callbacks'][$this->key($hook, $event)] ??=
+            array_reduce(
+                array_merge(
+                    [$event],
+                    is_object(Arr::first($payload)) ? $this->eventHierarchy(get_class(Arr::first($payload))) : []
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Arr' in method 'aggregateHierarchicalCallbacks'. (Clean Code Rules)

(StaticAccess)